### ISSUE TITLE
Old articles always on top

### ISSUE THREAD

**Issue by Megafry** [doublesecretagency/craft-viewcount#30]
My customer had the following feedback regarding the plugin:

New articles cannot appear on the list of most viewed because of the high view count older articles already have.
Therefore the plugins should add the option to make older articles less relevant over time.

One easy solution is to divide the view count by the age of the article.

**ex:**
Article 1: 1689 views, 3 days old  ➝ 563pt
Article 2: 23576 views, 308 days old ➝ 76pt
Article 3: 18232 views, 419 days old ➝ 43pt

**Or by months to only decrease the point from article order then 2 Months.** 
Article 1: 1689 views, 0 months ➝ 1689pt
Article 2: 23576 views, 3 months ➝ 7858pt
Article 3: 18232 views, 13 months ➝ 1402pt
Article 4: 501 views, 1 months ➝ 501pt

Would such an addition be possible?

best regards

**Comment by lindseydiloreto** [comment]
Thanks for the feedback, that's an interesting suggestion!

We have long-term plans which would make it easier to factor in the date of each view. No ETA, unfortunately.

In the meantime, you may be able to take advantage of [unique keys](https://plugins.doublesecretagency.com/view-count/using-a-unique-key/) to help break views into more distinct groups.

```twig
{# Generate unique key for this week #}
{% set thisWeek = now|date('W-Y') %}

{# Increment weekly counter #}
{% do craft.viewCount.increment(entry.id, thisWeek) %}
```

Hope that helps! We can leave this issue open for reference until a better long-term system is implemented.

**Comment by redburn** [comment]
I agree this would be helpful. In our case, for being able to rank the most popular user profiles in a give period. Grouping is an interesting approach but for a weekly tally I'd rather not have all the view counts start at zero at the start of the new week.

Instead, it would be great if I could filter the view counts by, say, views registered in last 7 days, like a rolling tally, but if I understand correctly that data is only available in the viewlog (if enabled), which isn't used to get the tally.

Since this is our main use case for this plugin, I'll have to look at other approaches like querying our analytics API.

(I now see this is also mentioned specifically in issue #25. Any update on the plans?)
